docs/marketplace/dev/front-end/table-design/index.md

```diff
@@ -0,0 +1,361 @@
+            search: () => import('@spryker/table.feature.search').then((m) => m.TableSearchFeatureModule),
+            syncStateUrl: () => import('@spryker/table.feature.sync-state').then((m) => m.TableSyncStateFeatureModule),
+            total: () => import('@spryker/table.feature.total').then((m) => m.TableTotalFeatureModule),
+            itemSelection: () =>
+                import('@spryker/table.feature.selectable').then((m) => m.TableSelectableFeatureModule),
+            batchActions: () =>
+                import('@spryker/table.feature.batch-actions').then((m) => m.TableBatchActionsFeatureModule),
+            columnConfigurator: () =>
+                import('@spryker/table.feature.settings').then((m) => m.TableSettingsFeatureModule),
+            title: () => import('@spryker/table.feature.title').then((m) => m.TableTitleFeatureModule),
+            editable: () => import('@spryker/table.feature.editable').then((m) => m.TableEditableFeatureModule),
+        }),
+    ],
+    providers: [
+        {
+            provide: TableDefaultConfig,
+            useClass: TableDefaultConfigData,
+        },
+    ],
+})
+export class DefaultTableConfigModule {}
+```
+
+### Columns
+
+Columns in a Table are defined by the `Column Type` and are rendered withing the columns (text, image, link, etc.).
+New Column Type may be created and registered to the Table.
+
+To use Column components, Table Module must implement a specific interface (TableConfig) and then be registered to the Root Module via TableModule.withColumnComponents():
+
+```ts
+class TableDefaultConfigData implements Partial<TableConfig> {
+    total = {
+        enabled: true,
+    };
+    columnConfigurator = {
+        enabled: true,
+    };
+}
+
+@NgModule({
+    imports: [
+        TableModule.forRoot(),
+        TableModule.withColumnComponents({
+            text: TableColumnTextComponent,
+            image: TableColumnImageComponent,
+            date: TableColumnDateComponent,
+            chip: TableColumnChipComponent,
+            input: TableColumnInputComponent,
+            select: TableColumnSelectComponent,
+            dynamic: TableColumnDynamicComponent,
+            autocomplete: TableColumnAutocompleteComponent,
+        } as any),
+
+        // Table Column Type Modules
+        TableColumnChipModule,
+        TableColumnTextModule,
+        TableColumnImageModule,
+        TableColumnDateModule,
+        TableColumnInputModule,
+        TableColumnSelectModule,
+        TableColumnDynamicModule,
+        TableColumnAutocompleteModule,
+    ],
+    providers: [
+        {
+            provide: TableDefaultConfig,
+            useClass: TableDefaultConfigData,
+        },
+    ],
+})
+export class DefaultTableConfigModule {}
+```
+
+### Actions 
+
+There is a way to trigger some [Actions](/docs/marketplace/dev/front-end/ui-components-library/actions.html) by the user interaction with the Table.
```

Comment:
Please add info that there are Table Features that can trigger actions like Row Actions Feature and Batch Actions Feature.